libimage/search.go

```diff
@@ -110,17 +114,17 @@ func (r *Runtime) Search(ctx context.Context, term string, options *SearchOption
 		options = &SearchOptions{}
 	}
 
-	var searchRegistries []string
+	searchRegistries := options.Registries
```

Comment:
Absolutely non-blocking: This is a bit far and non-obvious. (E.g. `if len(options.Registries) == 0 { } else { not use options.Registries! }` right now)

I’d prefer a very explicit
```go
var searchRegistries []string
if …; len(spa) > 1 { searchRegistries = … }
else if len(options.Registries) != 0 { searchRegistries = options.Registries }
else { searchRegistries = theDefault }
```
where just reading the code from top to bottom clearly shows which value is used, and I’d be happy to pay the three extra lines of code for this clarity.

Alternatively, leave it as is, but consider moving the initialization line above, to instead be just after the long comment below, and above the fist `if`. That doesn’t change the coding approach, it would just require a bit less short-term memory to understand.

Entirely up to you.

Reply:
Updated. The linter wanted it to be a switch.